sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/dataset.py

```diff
@@ -123,6 +160,46 @@ def __getitem__(self, index: int) -> BertSample:
             "is_random": torch.zeros_like(tokenized_sequence, dtype=torch.int64),
         }
 
+    def _tokenize_with_special_tracking(self, sequence: str) -> tuple[Tensor, list[int]]:
+        """Tokenize sequence and return positions where special tokens were added.
+
+        Args:
+            sequence: The protein sequence to tokenize
+
+        Returns:
+            Tuple of (tokenized_sequence, special_token_positions)
+        """
+        # Tokenize without special tokens to get original length
+        tokens_no_special = self.tokenizer.encode(sequence, add_special_tokens=False, return_tensors="pt")
+        if isinstance(tokens_no_special, list):
+            tokens_no_special = torch.tensor(tokens_no_special)
+        tokens_no_special = tokens_no_special.flatten()
+
+        # Tokenize with special tokens
+        tokens_with_special = self.tokenizer.encode(sequence, add_special_tokens=True, return_tensors="pt")
+        if isinstance(tokens_with_special, list):
+            tokens_with_special = torch.tensor(tokens_with_special)
+        tokens_with_special = tokens_with_special.flatten()
+
+        # Find positions where special tokens were inserted
```

Comment:
Is there an instance where special tokens are not at either end of the sequence?